File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/adapter/TabsListAdapter.java
package com.phlox.tvwebbrowser.activity.main.adapter;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import com.phlox.tvwebbrowser.activity.main.view.WebTabItemView;
import com.phlox.tvwebbrowser.model.WebTabState;

import java.util.List;

/**
 * Created by PDT on 24.08.2016.
 */
public class TabsListAdapter extends BaseAdapter {
    private List<WebTabState> states;
    private WebTabItemView.Listener tabsEventsListener;

    public TabsListAdapter(List<WebTabState> states, WebTabItemView.Listener tabsEventsListener) {
        this.states = states;
        this.tabsEventsListener = tabsEventsListener;
    }

    @Override
    public int getCount() {
        return states.size();
    }

    @Override
    public Object getItem(int i) {
        return states.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View convertView, ViewGroup viewGroup) {
        WebTabItemView view;
        if (convertView != null) {
            view = (WebTabItemView) convertView;
        } else {
            view = new WebTabItemView(viewGroup.getContext());
            view.setListener(tabsEventsListener);
        }
        view.bindTabState(states.get(i));
        return view;
    }
}

File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/adapter/TabsListAdapter.kt
package com.phlox.tvwebbrowser.activity.main.adapter

import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

import com.phlox.tvwebbrowser.activity.main.view.WebTabItemView
import com.phlox.tvwebbrowser.model.WebTabState

/**
 * Created by PDT on 24.08.2016.
 */
class TabsListAdapter(private val states: List<WebTabState>, private val tabsEventsListener: WebTabItemView.Listener) : BaseAdapter() {

    override fun getCount(): Int {
        return states.size
    }

    override fun getItem(i: Int): Any {
        return states[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(i: Int, convertView: View?, viewGroup: ViewGroup): View {
        val view: WebTabItemView
        if (convertView != null) {
            view = convertView as WebTabItemView
        } else {
            view = WebTabItemView(viewGroup.context)
            view.setListener(tabsEventsListener)
        }
        view.bindTabState(states[i])
        return view
    }
}
